src/pubsub/src/publisher/client.rs

```diff
@@ -159,6 +157,60 @@ impl Publisher {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct PublisherBuilder {
+    pub(crate) inner: crate::generated::gapic_dataplane::client::Publisher,
+    topic: String,
+    batching_settings: BatchingSettings,
+}
+
+impl PublisherBuilder {
+    /// Creates a new Pub/Sub publisher builder for topic.
+    pub(crate) fn new(
+        client: crate::generated::gapic_dataplane::client::Publisher,
+        topic: String,
+    ) -> Self {
+        Self {
+            inner: client,
+            topic,
+            batching_settings: BatchingSettings::default(),
+        }
+    }
+
+    // Change the message batching settings.
```

Comment:
For public documentation, we need three slashes `///`.

Also, these functions should have examples. Your plan was probably to defer writing the examples until we have the exact interfaces we want. That makes sense to me.